app/models/reservation.rb

```diff
@@ -25,7 +25,7 @@ class Reservation < ActiveRecord::Base
   validate :not_in_past, :available, :check_banned, on: :create
 
   # correctly update the overdue flag if necessary
-  before_save :update_overdue, if: :checked_out?
+  before_save :update_overdue, if: :checked_out?, unless: self.changes.empty?
```

Comment:
I don't know if compound conditions like this works, I'd probably just remove the `if` condition entirely and use guard clauses in the callback instead. What error shows up in the console when you try to start Rails?